src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -2265,9 +2265,44 @@ public void AddWorkspace(WorkspaceModel workspace)
             };
 
             _workspaces.Add(workspace);
+            //TODO move this to some handler for the event...
+            CheckForXMLDummyNodes(workspace);
             OnWorkspaceAdded(workspace);
         }
 
+        private void CheckForXMLDummyNodes(WorkspaceModel workspace)
+        {
+            //if the graph that is opened contains xml dummynodes log a notification 
+            if (workspace.containsXmlDummyNodes())
+            {
+                this.Logger.LogNotification("DynamoViewModel",
+                  Resources.UnresolvedNodesWarningTitle,
+                  Resources.UnresolvedNodesWarningShortMessage,
+                  Resources.UnresolvedNodesWarningMessage);
+
+                //raise a window as well so the user is clearly alerted to this state.
+                DisplayXmlDummyNodeWarning();
+            }
+        }
+        private void DisplayXmlDummyNodeWarning()
+        {
+           var xmlDummyNodeCount = this.CurrentWorkspace.Nodes.OfType<DummyNode>().
+                Where(node => node.OriginalNodeContent is XmlElement).Count();
+
+           Logging.Analytics.LogPiiInfo("XmlDummyNodeWarning",
+               xmlDummyNodeCount.ToString());
```

Comment:
Is this for Analytics so we know how often this happens for end user?

Reply:
that was my thought, yep.